File: 0169-majority-element/0169-majority-element.java
import java.util.*;
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> d = new HashMap<>();
        for(int i: nums)
        {
            if(d.containsKey(i))
            {
                d.put(i, d.get(i) +1);
            }
            else
            {
                d.put(i, 1);
            }
        }
        int max_elem = nums[0];
        int max_elem_count = 0;
        for (var entry : d.entrySet()) {
            if (entry.getValue() > max_elem_count) {
                max_elem_count = entry.getValue();
                max_elem = entry.getKey();
            }
        }
        return max_elem;
        
    }
}

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        Dictionary<int, int> d = new Dictionary<int, int>();
        foreach(int num in nums)
        {
            if( d.ContainsKey(num))
            {
                d[num] ++;
            }
            else 
            {
                d[num] = 1;
            }
        }
        var max_elem = nums[0];
        var max_elem_count = 0;
        foreach(var dict in d)
        {
            if(d[dict.Key] > max_elem_count)
            {
                max_elem = dict.Key;
                max_elem_count = d[dict.Key];
            }
        }
        return max_elem;
        
    }
}